runtime/server/catalog.go

```diff
@@ -61,6 +74,11 @@ func (s *Server) Reconcile(ctx context.Context, req *runtimev1.ReconcileRequest)
 
 // PutFileAndReconcile implements RuntimeService.
 func (s *Server) PutFileAndReconcile(ctx context.Context, req *runtimev1.PutFileAndReconcileRequest) (*runtimev1.PutFileAndReconcileResponse, error) {
+	claims := auth.GetClaims(ctx)
+	if !claims.CanInstance(req.InstanceId, auth.EditRepo) || !claims.CanInstance(req.InstanceId, auth.EditInstance) {
```

Comment:
So this is for when users can do modeling on cloud ? locally auth will be disabled and for cloud deployment whoever can push to git repo will be able to make changes. I believe this method is also called when runtime server starts then what permissions will be used. Will there be a system level admin user ?

Reply:
Yeah, this will be for modeling on cloud (which won't initially be available). I added these checks to all endpoints to prevent users from calling them in cloud (since we won't issue tokens for modeling in cloud initially). 

When the runtime starts, it only calls reconcile on local. In cloud, it will be the admin server that calls reconcile when it creates instances or gets a Git webhook that code files have changed.

When the admin server needs to call reconcile, it can use its `Issuer` to create a JWT that has the correct permissions to call reconcile :)